python/tests/rtl/test_user_session.py

```diff
@@ -0,0 +1,125 @@
+# pylint: disable=redefined-outer-name
+
+import pytest
+
+from looker.rtl.user_session import UserSession
+from looker.rtl.api_settings import ApiSettings
+from looker.rtl import transport as tp
+
+
+@pytest.fixture(scope='session')
+def config_file(tmpdir_factory):
+    """Creates a sample looker.ini file and returns it"""
+    filename = tmpdir_factory.mktemp('settings').join('looker.ini')
+    filename.write("""
+[Looker]
+# API version is required
+api_version=3.1
+# Base URL for API. Do not include /api/* in the url
+base_url=https://host1.looker.com:19999
+# API 3 client id
+client_id=your_API3_client_id
+# API 3 client secret
+client_secret=your_API3_client_secret
+# Optional embed secret for SSO embedding
+embed_secret=your_embed_SSO_secret
+# Optional user_id to impersonate
+user_id=
+# Set to false if testing locally against self-signed certs. Otherwise leave True
+verify_sl=True
+# leave verbose off by default
+verbose=false
+        """)
+    return filename
+
+
+# pylint: disable=no-method-argument
+class MockTransport():
```

Comment:
ah, I see you've followed my pattern :-) I thought you were going to experiment with pytest monkey patching but I'll admit I like this approach better. It might require a little more code but I find it more explicit and readable.